### PR TITLE
DX: Narrow docblock types in FixerConfiguration

### PR DESCRIPTION
**Pull request by jrmajor**

### CODE REVIEW

File: src/Console/Command/DescribeCommand.php [hunk 1]
```diff
@@ -196,19 +196,15 @@ private function describeRule(OutputInterface $output, string $name): void
...
                     $allowed = array_map(
-                        static function (string $type): string {
-                            return '<comment>'.$type.'</comment>';
-                        },
-                        $option->getAllowedTypes()
+                        static fn (string $type): string => '<comment>'.$type.'</comment>',
+                        $option->getAllowedTypes(),
                     );
                 } else {
-                    foreach ($allowed as &$value) {
```
Comment (jrmajor) [inline src/Console/Command/DescribeCommand.php:205]: Replaced with `array_map` as PHPStan was confused by the reference.

File: src/FixerConfiguration/FixerOption.php [hunk 2]
```diff
@@ -43,8 +43,9 @@ final class FixerOption implements FixerOptionInterface
     private $normalizer;
 
     /**
-     * @param mixed         $default
-     * @param null|string[] $allowedTypes
+     * @param mixed             $default
+     * @param null|list<string> $allowedTypes
+     * @param null|list<(callable(mixed): bool)|null|scalar> $allowedValues
```
Comment (julienfalque) [inline src/FixerConfiguration/FixerOption.php:48]: Looks like we have a missing case in `phpdoc_align` 😄

**Review by julienfalque — APPROVED** [review]

**Comment by julienfalque** [comment]
Thank you @jrmajor.

File: src/Console/Command/HelpCommand.php [hunk 3]
```diff
@@ -50,14 +50,16 @@ public static function toString($value): string
 
     /**
      * Returns the allowed values of the given option that can be converted to a string.
+     *
+     * @return list<null|AllowedValueSubset|scalar>
```
Comment (keradus) [inline src/Console/Command/HelpCommand.php:54]: but... method prototype says that it can return null?

Comment (jrmajor) [reply src/Console/Command/HelpCommand.php:54]: Sorry, that's a mistake (ref #6584).

**Pull request merged**